index.html

```diff
@@ -456,32 +421,171 @@ <h2 id="constraints">
+            </tr>
+            <tr id="def-constraint-frameRate">
+              <td><dfn>frameRate</dfn></td>
+              <td><code><a>ConstrainDouble</a></code></td>
+              <td>The frame rate (frames per second) or frame rate range. As a
+              capability, max MUST reflect the <a>display surface</a>'s frame
+              rate, and min MUST reflect the lowest frame rate available through
+              frame decimation by the user agent.</td>
+            </tr>
+            <tr id="def-constraint-aspect">
+              <td><dfn>aspectRatio</dfn></td>
+              <td><code><a>ConstrainDouble</a></code></td>
+              <td>The exact aspect ratio (width in pixels divided by height in
+              pixels, represented as a double rounded to the tenth decimal
+              place) or aspect ratio range. As a setting, represents the aspect
+              ratio of the <a>display surface</a>. As a capability, min and max
+              both MUST be the setting value, rendering this property immutable
+              from the application viewpoint. However, these values may change
+              if the end-user is able to resize the <a>display surface</a>.</td>
+            </tr>
+            <tr id="def-constraint-resizeMode">
+              <td><dfn>resizeMode</dfn></td>
+              <td><code><a>ConstrainDOMString</a></code></td>
+              <td>
+                This string (or each string, when a list) should be one of the
+                members of <code><a>VideoResizeModeEnum</a></code>. As a setting,
+                <code>none</code> means the <code><a>MediaStreamTrack</a></code>
+                contains all bits needed to render the display in full detail,
+                which if <code>window.devicePixelRatio > 1</code>, means
+                <code>width</code> and <code>height</code> will be larger
+                than the display's appearance from an end-user viewpoint would
+                suggest, whereas <code>crop-and-scale</code> means the
+                <code><a>MediaStreamTrack</a></code> contains an
+                aspect-preserved representation of the <a>display surface</a>
+                that has been downscaled by the user agent, but not cropped. As
+                a capability, the values <code>none</code> and
+                <code>crop-and-scale</code> both MUST be present. In other
+                words, the user agent MUST support downscaling.
+              </td>
+            </tr>
+            <tr id="def-constraint-displaySurface">
+              <td><dfn>displaySurface</dfn></td>
+              <td><code><a>ConstrainDOMString</a></code></td>
+              <td>
+                This string (or each string, when a list) should be one of the
+                members of <code><a>DisplayCaptureSurfaceType</a></code>. As a
+                setting, indicates the type of <a>display surface</a> that
+                is being captured. As a capability, the setting value MUST be
```

Comment:
Do we allow this changing (just like aspectRatio can change)? I think it should be allowed by the user agent to change what display surface is being captured in real-time, e.g. based on browser UI.

Reply:
By "changing" if you mean though applying constraints, then no, we do not allow applications to affect `aspectRatio`. It is *"immutable from the application viewpoint"*.

The spec already explicitly says `displaySurface` is read-only, and this PR is consistent with that.

I agree with the spec here, and I disagree applications should be able to change `displaySurface` since this undermines the model as exposed to users, and would be surprising.

In my view, a user agent that wants to implement some sort of "follow my active tab" exposure, would need to implement a singular "smart" virtual device (a `logicalSurface` if you will), and be consistent within this model (meaning: make it a discrete choice in the UA selector, and preferably explain it well to the user).

Personally, I question the security of letting JS applications peruse one's tabs.